airflow-core/src/airflow/ui/src/queries/useRefreshOnNewDagRuns.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { useQueryClient } from "@tanstack/react-query";
+import { useEffect, useRef } from "react";
+
+import {
+  useDagRunServiceGetDagRuns,
+  useDagServiceGetDagDetailsKey,
+  UseDagRunServiceGetDagRunsKeyFn,
+  UseDagServiceGetDagDetailsKeyFn,
+  useDagServiceGetDagsKey,
+  useDagsServiceRecentDagRunsKey,
+  UseGridServiceGridDataKeyFn,
+  UseTaskInstanceServiceGetTaskInstancesKeyFn,
+} from "openapi/queries";
+
+import { useConfig } from "./useConfig";
+
+export const useRefreshOnNewDagRuns = (dagId: string, hasPendingRuns: boolean | undefined) => {
+  const queryClient = useQueryClient();
+  const previousDagRunIdRef = useRef<string>();
+  const autoRefreshInterval = useConfig("auto_refresh_interval") as number;
+
+  const { data } = useDagRunServiceGetDagRuns({ dagId, limit: 1, orderBy: "-run_after" }, undefined, {
+    enabled: Boolean(dagId),
+    refetchInterval: hasPendingRuns
+      ? false
+      : Boolean(autoRefreshInterval)
+        ? autoRefreshInterval * 1000
+        : 5000,
+  });
+
+  useEffect(() => {
+    const latestDagRun = data?.dag_runs[0];
+
+    const latestDagRunId = latestDagRun?.dag_run_id;
+
+    if ((latestDagRunId ?? "") && previousDagRunIdRef.current !== latestDagRunId) {
+      previousDagRunIdRef.current = latestDagRunId;
+
+      const queryKeys = [
+        [useDagServiceGetDagsKey],
+        [useDagsServiceRecentDagRunsKey],
+        [useDagServiceGetDagDetailsKey],
+        UseDagServiceGetDagDetailsKeyFn({ dagId }, [{ dagId }]),
```

Comment:
```suggestion
        [useDagsServiceRecentDagRunsKey],
```

Let's only invalidate queries based on dag runs or task instances.

Reply:
This all queries are similar to when we trigger a dag run from the dag details page, except  `useDagServiceGetDagDetailsKey` which takes care for other details such as dag pause.